cpp/src/gandiva/CMakeLists.txt

```diff
@@ -36,6 +36,7 @@ provide_cmake_module(GandivaAddBitcode "Gandiva")
 
 # Set the path where the bitcode file generated, see precompiled/CMakeLists.txt
 set(GANDIVA_PRECOMPILED_BC_PATH "${CMAKE_CURRENT_BINARY_DIR}/irhelpers.bc")
+set(GANDIVA_PRECOMPILED_MANDATORY_BC_PATH "${CMAKE_CURRENT_BINARY_DIR}/mandatory_ir.bc")
 set(GANDIVA_PRECOMPILED_CC_PATH "${CMAKE_CURRENT_BINARY_DIR}/precompiled_bitcode.cc")
```

Comment:
There are two bitcode files now, the original `irhelpers.bc` is optionally used, and the `mandatory_ir.bc` is mandatory and always linked to the LLVM module constructed.